sbpy/photometry/core.py

```diff
@@ -551,52 +573,142 @@ def phaseint(self):
         """Phase integral"""
         return self._phase_integral()
 
-    def fit(self, eph):
-        """Fit photometric model to photometric data stored in sbpy.data.Ephem
-        object
+    def fit(self, eph, mag, fitter=None, return_fitter=False, **kwargs):
+        """Fit disk-integrated phase function model to magnitude data and the
+        corresponding ephemerides data.
+
+        This is a wrapper that provides a consistent interface that is
+        compatible with the `sbpy` guideline.
 
         Parameters
         ----------
-        eph : `~sbpy.data.Ephem` instance, mandatory
-            photometric data; must contain `phase` (phase angle) and `mag`
-            (apparent magnitude) columns; `mag_err` optional
+        eph : `~sbpy.data.Ephem`, dict_like, or array_like of float
+            If `~sbpy.data.Ephem` or dict_like, must contain a column with a
+            name that's compatible with the alternative names for phase angle
+            (see `~sbpy.data.DataClass`).  If array_like, then it is the phase
+            angles of object.  If any distance (heliocentric and geocentric)
+            is provided, then they will be used to correct magnitude to
+            1 au before fitting.  If no unit is provided via type
+            `~astropy.units.Quantity`, then radiance is assumed for phase
```

Comment:
This is because the base functions in the IAU H, G and other systems are defined in radians.  I can set degrees as the default unit for angles and add the degree-to-radians conversion into `.evaluate` and `fit_deriv`, but then the code will run nearly 4x slower when the input angle is dimensionless, and about 1.5x slower for Quantity input.

So what's your suggestion for this trade off?

Reply:
How about making a copy of the `Ephem` object and converting the phase column to radians?

Reply:
We could do this, but then the question is, do we want to do this for all methods, including overloading the `.__call__` method?  If we do so, then there will be inconsistency and confusion from attribute `.input_units` because it specifies the default units taken by the static method `.evaluate` as defined by the base class `astropy.modeling.Model`...

Reply:
The point of this method is to provide a convenience for the user with `sbpy` classes, right?  Seems like we should be requiring something that is compatible with the `Ephem` class.  That class should require units for these parameters.  What the other methods do can be treated separately, right?

Reply:
Can you make `evaluate` and all `astropy.modeling` aspects work with radians and then do a conversion in the higher level methods that the user is calling? It's not really the cleanest solution, but I think users would really prefer to work with angles instead of radians...

Reply:
@mkelley we are trying to make as many methods as possible to work with `sbpy` classes, so it's not just a few methods that we treat differently.  But if we change the default of all methods, then we might need to touch the core of `astropy.modeling.Model`, which we should definitely avoid.

@mommermi True that I like to work in degrees rather than radians.  But I'm really worried that different default unit for different methods will potentially create serious confusions.  In my opinion, setting default as 'rad' isn't a big issue for users, because all methods (including those `astropy.modeling.Model` methods) can take `Quantity` input, and the `Ephem` object, at least when generated from online sources using, e.g., `.from_horizons` all have phase angles in `Quantity`.  If users are not sure, they can just explicitly supply a unit to phase angle.  On the other hand, leaving default as 'rad' keeps the code clean.  So my preference is still let 'rad' as default.